src/components/MessagesList/MessagesGroup/Message/MessagePart/ReactionsList.vue

```diff
@@ -0,0 +1,191 @@
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<!-- all reactions -->
+<template>
+	<NcModal size="small"
+		:container="container"
+		@close="closeModal">
+		<div class="reactions__modal">
+			<h2>
+				{{ t('spreed', 'Reactions') }}
+			</h2>
+			<template v-if="Object.keys(detailedReactions).length > 0">
+				<div class="reactions-list__navigation">
+					<NcButton :class="{'active' : reactionFilter === ''}"
+						type="tertiary"
+						@click="handleTabClick('')">
+						{{ t('spreed', 'All ({count})', { count: flatReactions.length }) }}
+					</NcButton>
+					<NcButton v-for="reaction in reactionsSorted"
+						:key="reaction"
+						:class="{'active' : reactionFilter === reaction}"
+						type="tertiary"
+						@click="handleTabClick(reaction)">
+						{{ reaction }} {{ detailedReactions[reaction].length }}
+					</NcButton>
+				</div>
+
+				<NcListItemIcon v-for="item in filteredReactions ?? flatReactions"
+					:key="item.actorDisplayName"
+					:name="getDisplayNameForReaction(item)">
+					<span>
+						{{ reactionFilter === '' ? item.reaction.join('') : reactionFilter }}
+					</span>
+				</NcListItemIcon>
+			</template>
+			<NcLoadingIcon v-else :size="64" />
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcListItemIcon from '@nextcloud/vue/dist/Components/NcListItemIcon.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+
+import { ATTENDEE } from '../../../../../constants.js'
+import { useGuestNameStore } from '../../../../../stores/guestName.js'
+
+export default {
+
+	name: 'ReactionsList',
+
+	components: {
+		NcModal,
+		NcLoadingIcon,
+		NcListItemIcon,
+		NcButton,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+
+		detailedReactions: {
+			type: Object,
+			default: () => {},
+		},
+
+		reactionsSorted: {
+			type: Array,
+			default: () => [],
+		},
+	},
+
+	emits: ['close'],
+
+	setup() {
+		return {
+			guestNameStore: useGuestNameStore(),
+		}
+	},
+
+	data() {
+		return {
+			reactionFilter: '',
+		}
+	},
+
+	computed: {
+		container() {
+			return this.$store.getters.getMainContainerSelector()
+		},
+
+		flatReactions() {
```

Comment:
Consider (to reduce computations) storing of flatReactions as object:
```js
{
  'all': [Array],
  😆 : [Array],
  ...,
}
``` 
then you don't  need 'filteredReactions' prop, as you can access values by key 'reactionFilter'

Reply:
It will increase stored data instead which is not needed as it is redundant ( 'all' includes a+b+c+d, emoji1 includes a, emoji2 includes b etc)

Reply:
I thought more about computing = iterating over the array
With M emojis, generating and checking all of them will iterate M+1 times, plus array.join for each person in All tab

Reply:
You are not wrong ...